src/graph/expression_graph.h

```diff
@@ -112,97 +123,167 @@ class Tensors {
 
   std::unordered_map<size_t, std::vector<Expr>> memoized_;
 
-  Type defaultElementType_{Type::float32}; // Type used for storing parameters, currently all parameters have to have the same type
+  Type defaultElementType_{Type::float32}; /**< Type used for storing parameters, currently all parameters have to have the same type */
 
-  bool inferenceOnly_{false};
+  bool inferenceOnly_{false}; /**< a flag holds whether the graph is used for inference only */
 
-  bool checkpointing_{false}; // use gradient checkpointing if true
+  bool checkpointing_{false}; /**< use gradient checkpointing if true */
 
-  bool reloaded_{false};
+  bool reloaded_{false}; /**< a flag holds whether the graph is reloaded: reloaded is true if the graph loads parameters by load() function. */
 
-  bool throwNaN_{false};
+  bool throwNaN_{false}; /**< a flag holds whether the graph throws a NaN exception */
 
 protected:
   // Delete, copy and move constructors
   ExpressionGraph(const ExpressionGraph&) = delete;
   ExpressionGraph(ExpressionGraph&&) = delete;
 
-  // Holds memory and expressions that correspond to graph parameters
-  // Now we can have multiple types of parameters in a separate parameters object per value type. 
-  // This is currently only accessible through private functions during loading, will abort during training
-  // when params() is called (e.g. optimizer) and there is more or other types than the default parameter type.
-  // Currently the only usecase is inference. Trying to access params() for non-default parameter type is going
-  // to abort. Inference does not need to access a whole set of parameters.
+  /**
+   * @brief A map holds memory and nodes that corresponds to graph parameters
+   *
+   * The key is Type and the mapped value is a set of parameter objects with corresponding type.
+   * Now we can have multiple types of parameters in a separate parameters object per value type.
+   * This is currently only accessible through private functions during loading, will abort during training
+   * when params() is called (e.g. optimizer) and there is more or other types than the default parameter type.
+   * Currently the only usecase is inference. Trying to access params() for non-default parameter type is going
+   * to abort. Inference does not need to access a whole set of parameters.
+   */
   ElementTypeParamsMap paramsByElementType_;
-  Ptr<Backend> backend_;
-
-  std::string namespace_;
+  Ptr<Backend> backend_;  /**< @brief a pointer to the backend for the graph */
+  std::string namespace_; /**< @brief a string defines the namespace of the graph. Each graph has its own unique namespace. */
 
 public:
-  /** @brief Constructs a new expression graph
+  /**
+   * @brief Constructs a new expression graph
    *
    * Constructor should be used as New<ExpressionGraph>()
    */
   ExpressionGraph(bool inference = false);
 
+  /**
+   * @brief Destructor
+   *
+   * Clear everything related to the graph except memoized nodes
+   */
```

Comment:
I wonder if with using automatic `@brief`ing in doxygen, we could reduce this docs to just a single line.